src/Castle.Core/DynamicProxy/Internal/InvocationHelper.cs

```diff
@@ -50,7 +50,7 @@ public static MethodInfo GetMethodOnType(Type type, MethodInfo proxiedMethod)
 
 			var cacheKey = new CacheKey(proxiedMethod, type);
 
-			return cache.GetOrAdd(cacheKey, ck => ObtainMethod(proxiedMethod, type));
+			return cache.GetOrAdd(cacheKey, static ck => ObtainMethod(ck.Method, ck.Type));
```

Comment:
(Theoretically generated code could be reduced by changing `ObtainMethod` such that it accepts a single `CacheKey` argument, then it could be used directly here in place of the lambda function wrapping it. But that's a different refactoring perhaps for another time.)

Reply:
Do you mean method group? I don't mind doing that if you like.

Reply:
I meant replacing the whole `ck => ObtainMethod(ck.Method, ck.Type)` with just `ObtainMethod`, which might be possible if that method were made to accept a single`CacheKey` argument instead of two. But I just mentioned in it passing, no need to change that.